src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -57,13 +57,35 @@ private sealed partial class SymbolStartAnalyzer
                 var deserializationConstructorCheck = new DeserializationConstructorCheck(context.Compilation);
                 context.RegisterSymbolStartAction(symbolStartContext =>
                 {
+                    if (HasSyntaxErrors((INamedTypeSymbol)symbolStartContext.Symbol, symbolStartContext.CancellationToken))
+                    {
+                        // Bail out on syntax errors.
+                        return;
+                    }
+
                     // Create a new SymbolStartAnalyzer instance for every named type symbol
                     // to ensure there is no shared state (such as identified unused parameters within the type),
                     // as that would lead to duplicate diagnostics being reported from symbol end action callbacks
                     // for unrelated named types.
                     var symbolAnalyzer = new SymbolStartAnalyzer(analyzer, eventsArgType, attributeSetForMethodsToIgnore, deserializationConstructorCheck);
                     symbolAnalyzer.OnSymbolStart(symbolStartContext);
                 }, SymbolKind.NamedType);
+
+                return;
+
+                // Local functions
+                static bool HasSyntaxErrors(INamedTypeSymbol namedTypeSymbol, CancellationToken cancellationToken)
+                {
+                    foreach (var tree in namedTypeSymbol.Locations.Select(l => l.SourceTree))
+                    {
+                        if (tree.GetDiagnostics(cancellationToken).Any(d => d.Severity == DiagnosticSeverity.Error))
```

Comment:
There's no node we can grab and call SyntaxNode.ContainsDiagnostics?

Reply:
Unfortunately 'containsdiagnostic' is often wrong.  For example, syntactic *warnings* will trigger it.  

We should just get the SyntaxReferences for the NamedTypeSymbol though and check those directly (i.e. checking for the right severity).  

so this should be more like:

```c#
foreach (var syntaxRef in namedTypeSymbol.DeclaringSyntaxReferences) 
     if (syntaxRef.GetSyntax(cancellationToken).GetDiagnostics(...))`